spyrograph/_trochoid.py

```diff
@@ -67,7 +67,13 @@ def __init__(
         self.d = d
         self.thetas = self._validate_theta(thetas, theta_start, theta_stop, theta_step)
         self.origin = origin
-
+   
+        while deci != False:
```

Comment:
Almost there! All we have to do now is remove this `while` loop as we don't have to loop on this particular spot

Also `deci` isn't defined prior to line 71 so this would cause the `while` loop to fail anyway

All we need are lines 72 and 73 so this whole block could just be replaced with 

```python
if self.R <= 0 or self.r <= 0 or self.d <= 0:
    raise ValueEror("Negative and/or zero input parameters were passed. Please only pass positive values")
```

Reply:
What if the user inputs the wrong value again?
I put in a loop thinking about that.

Reply:
The program itself will fail upon raising the error so the developer writing the code would modify the code and then rerun

Reply:
I am making such silly mistakes 😞

Reply:
For example say I write some code that has a bug

My program will raise an error so I would then rewrite the code to fix the error and then rerun the program

Reply:
Nono!! Don't worry at all - this is a great learning opportunity :D

Reply:
Ohk understood.
I don't have any prior experience using ValueError.

Reply:
We all start somewhere - you're doing great!! I hugely appreciate your help on this and we're almost there! Just this last fix and I'll merge :~)